src/NonLinearProgram/nlp_utilities.jl

```diff
@@ -0,0 +1,556 @@
+    primal_idx = [i.value for i in model.cache.primal_vars]
+    X = [model.x[primal_idx]; s_leq; s_geq]
+
+    # value and dual of the lower bounds
+    V_L = spzeros(num_vars + num_ineq)
+    X_L = spzeros(num_vars + num_ineq)
+    for (i, j) in enumerate(has_low)
+        V_L[j] =
+            model.y[form.constraint_lower_bounds[primal_vars[j].value].value] *
+            sense_multiplier
+        #dual.(LowerBoundRef(primal_vars[j])) * sense_multiplier
+        #
+        if sense_multiplier == 1.0
+            V_L[j] <= -tol &&
+                @info "Dual of lower bound must be positive" i V_L[j]
+        else
+            V_L[j] >= tol &&
+                @info "Dual of lower bound must be negative" i V_L[j]
+        end
+        #
+        X_L[j] = form.lower_bounds[primal_vars[j].value]
+    end
+    for (i, con) in enumerate(cons[geq_locations])
+        # By convention jump dual will allways be positive for geq constraints
+        # but for ipopt it will be positive if min problem and negative if max problem
+        V_L[num_vars+i] =
+            model.y[form.nlp_index_2_constraint[con].value] * sense_multiplier
+        #
+        if sense_multiplier == 1.0
+            V_L[num_vars+i] <= -tol &&
+                @info "Dual of geq constraint must be positive" i V_L[num_vars+i]
+        else
+            V_L[num_vars+i] >= tol &&
+                @info "Dual of geq constraint must be negative" i V_L[num_vars+i]
+        end
+    end
+    # value and dual of the upper bounds
+    V_U = spzeros(num_vars + num_ineq)
+    X_U = spzeros(num_vars + num_ineq)
+    for (i, j) in enumerate(has_up)
+        V_U[j] =
+            model.y[form.constraint_upper_bounds[primal_vars[j].value].value] *
+            (-sense_multiplier)
+        # dual.(UpperBoundRef(primal_vars[j])) * (- sense_multiplier)
+        if sense_multiplier == 1.0
+            V_U[j] <= -tol &&
+                @info "Dual of upper bound must be positive" i V_U[i]
+        else
+            V_U[j] >= tol &&
+                @info "Dual of upper bound must be negative" i V_U[i]
+        end
+        #
+        X_U[j] = form.upper_bounds[primal_vars[j].value]
+    end
+    for (i, con) in enumerate(cons[leq_locations])
+        # By convention jump dual will allways be negative for leq constraints
+        # but for ipopt it will be positive if min problem and negative if max problem
+        V_U[num_vars+num_geq+i] =
+            model.y[form.nlp_index_2_constraint[con].value] *
+            (-sense_multiplier)
+        # dual.(con) * (- sense_multiplier)
+        if sense_multiplier == 1.0
+            V_U[num_vars+num_geq+i] <= -tol &&
+                @info "Dual of leq constraint must be positive" i V_U[num_vars+i]
+        else
+            V_U[num_vars+num_geq+i] >= tol &&
+                @info "Dual of leq constraint must be negative" i V_U[num_vars+i]
+        end
+    end
+    return X,
```

Comment:
I would maybe do a pass of JuliaFormatters there

Reply:
I did, this was all done by formatter